operator/api/v1alpha1/keptnworkload_types.go

```diff
@@ -77,3 +80,33 @@ func init() {
 func (w KeptnWorkload) GetWorkloadInstanceName() string {
 	return strings.ToLower(w.Name + "-" + w.Spec.Version)
 }
+
+func (v KeptnWorkload) SetSpanAttributes(span trace.Span) {
+	attributes := v.GetSpanAttributes()
+	for _, attribute := range attributes {
+		span.SetAttributes(attribute)
+	}
+}
+
+func (v KeptnWorkload) GenerateWorkloadInstance(previousVersion string, traceContextCarrier map[string]string) KeptnWorkloadInstance {
+	return KeptnWorkloadInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: traceContextCarrier,
+			Name:        v.GetWorkloadInstanceName(),
+			Namespace:   v.Namespace,
+		},
+		Spec: KeptnWorkloadInstanceSpec{
+			KeptnWorkloadSpec: v.Spec,
+			WorkloadName:      v.Name,
+			PreviousVersion:   previousVersion,
+		},
+	}
+}
+
+func (i KeptnWorkload) GetSpanAttributes() []attribute.KeyValue {
```

Comment:
let's use the same receiver name in each function (currently we use i, v and w) - I suggest `w`